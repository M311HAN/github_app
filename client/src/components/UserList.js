import React from "react";
import { Link } from "react-router-dom";

const UserList = ({ users }) => {
  // If no users are found, display a message
  if (users.length === 0) {
    return <p className="loading-text">No users found</p>;
  }

  return (
    <div className="container mt-5">
      <div className="row">
        {users.map((user) => (
          <div className="col-md-4 mb-4" key={user.id}>
            <div className="card h-100 shadow-sm">
              <img
                src={user.avatar_url}
                alt={`Avatar of ${user.login}`}
                className="card-img-top img-fluid"
                style={{
                  height: "150px",
                  objectFit: "cover",
                  borderRadius: "50%",
                  margin: "20px auto",
                  width: "150px",
                }}
              />
              <div className="card-body text-center">
                <h5 className="card-title">{user.login}</h5>
                <p className="card-text">ID: {user.id}</p>
                <a
                  href={user.html_url}
                  className="btn btn-success mb-2"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  GitHub Profile
                </a>
                <Link
                  to={`/users/${user.login}`}
                  className="btn btn-link text-primary"
                >
                  User's Page
                </Link>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default UserList;
