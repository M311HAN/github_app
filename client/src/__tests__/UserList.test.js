import React from "react";
import { render } from "@testing-library/react";
// Import Router to handle routing within the test
import { BrowserRouter as Router } from "react-router-dom";
// Import the UserList component to be tested
import UserList from "../components/UserList";

// Sample user data to be passed as a prop to UserList
const users = [
  {
    id: 1,
    login: "octocat",
    avatar_url: "https://avatars.githubusercontent.com/u/583231?v=4",
  },
  {
    id: 2,
    login: "user2",
    avatar_url: "https://avatars.githubusercontent.com/u/2?v=4",
  },
];

test("renders correctly", () => {
  // Render the UserList component within a Router, passing in the sample user data
  const { asFragment } = render(
    <Router>
      <UserList users={users} onSelectUser={() => {}} />
    </Router>
  );
  // Check that the rendered component matches the previous snapshot
  expect(asFragment()).toMatchSnapshot();
});
