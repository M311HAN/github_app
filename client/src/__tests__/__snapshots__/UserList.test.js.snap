// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="row"
    >
      <div
        class="col-md-4 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Avatar of octocat"
            class="card-img-top img-fluid"
            src="https://avatars.githubusercontent.com/u/583231?v=4"
            style="height: 150px; object-fit: cover; border-radius: 50%; margin: 20px auto; width: 150px;"
          />
          <div
            class="card-body text-center"
          >
            <h5
              class="card-title"
            >
              octocat
            </h5>
            <p
              class="card-text"
            >
              ID: 1
            </p>
            <a
              class="btn btn-success mb-2"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub Profile
            </a>
            <a
              class="btn btn-link text-primary"
              href="/users/octocat"
            >
              User's Page
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-md-4 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Avatar of user2"
            class="card-img-top img-fluid"
            src="https://avatars.githubusercontent.com/u/2?v=4"
            style="height: 150px; object-fit: cover; border-radius: 50%; margin: 20px auto; width: 150px;"
          />
          <div
            class="card-body text-center"
          >
            <h5
              class="card-title"
            >
              user2
            </h5>
            <p
              class="card-text"
            >
              ID: 2
            </p>
            <a
              class="btn btn-success mb-2"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub Profile
            </a>
            <a
              class="btn btn-link text-primary"
              href="/users/user2"
            >
              User's Page
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
